src/daft-core/src/array/ops/full.rs

```diff
@@ -25,6 +25,12 @@ where
 {
     /// Creates a DataArray<T> of size `length` that is filled with all nulls.
     fn full_null(name: &str, dtype: &DataType, length: usize) -> Self {
+        if dtype != &T::get_dtype() {
```

Comment:
@jaychia added this check for dtype in `full_null` case.
Now it panics with wrong data type

Reply:
We might actually also need to add an additional clause here that `!matches!(T::get_dtype(), DataType::Unknown)`. because some datatypes (in this case, potentially our Extension types) will return `DataType::Unknown` as these types cannot be determined at static-time.
```suggestion
        if dtype != &T::get_dtype() && !matches!(T::get_dtype(), DataType::Unknown) {
```